examples/multi-agent-debate-system/main.py

```diff
@@ -0,0 +1,299 @@
+import typer
+from rich.prompt import Prompt
+import langroid as lr
+from agents import (
+    create_pro_ai_agent,
+    create_con_ai_agent,
+    create_pro_ip_agent,
+    create_con_ip_agent,
+    create_pro_bias_agent,
+    create_con_bias_agent,
+    create_pro_edu_agent,
+    create_con_edu_agent,
+    create_feedback_agent,
+)
+from tasks import (
+    create_pro_ai_task,
+    create_con_ai_task,
+    create_pro_ip_task,
+    create_con_ip_task,
+    create_pro_bias_task,
+    create_con_bias_task,
+    create_pro_edu_task,
+    create_con_edu_task,
+    create_feedback_task,
+)
+from config import (
+    get_global_settings,
+    get_base_llm_config,
+    get_llm_delegate_setting,
+)
+import logging
+
+# Set up the Typer application
+app = typer.Typer()
+
+# Set up logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+lr.utils.logging.setup_colored_logging()
+
+
+
+# === SETUP AGENTS AND TASKS ===
+def setup_agents_and_tasks(debug=False, nocache=False):
+    """
+    Sets up agents and tasks for the debate.
+
+    Args:
+        debug (bool): Enable or disable debug mode.
+        nocache (bool): Enable or disable caching.
+
+    Returns:
+        dict: A dictionary of tasks mapped by their names.
+    """
+    logger.info("Setting up agents and tasks...")
+    global_settings = get_global_settings(debug, nocache)
+    lr.utils.configuration.set_global(global_settings)
+    logger.info("Global settings configured. Debug=%s, Cache=%s", debug, not nocache)
+
+    base_llm_config = get_base_llm_config()
+    logger.info("Base LLM configuration obtained.")
+
+
+    # Create agents and reset memory for LLMS using the functions from agents.py
+    #AI in Healthcare Pro and Con Agents
+    pro_ai_agent = create_pro_ai_agent(base_llm_config)
+    pro_ai_agent.clear_history(0)
```

Comment:
you can use `agent.init_state()` which clears both histories. Also `dialog` is not used unless it's a `DocChatAgent` (I think) so you only really need to do `clear_history`